src/main/java/dev/hephaestus/atmosfera/client/sound/modifiers/implementations/BossBarCondition.java

```diff
@@ -0,0 +1,55 @@
+package dev.hephaestus.atmosfera.client.sound.modifiers.implementations;
+
+import com.google.gson.JsonObject;
+import dev.hephaestus.atmosfera.client.sound.modifiers.AtmosphericSoundModifier;
+import dev.hephaestus.atmosfera.world.context.EnvironmentContext;
+import net.minecraft.client.gui.hud.ClientBossBar;
+import net.minecraft.entity.boss.BossBar;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.world.World;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.regex.Pattern;
+
+public record BossBarCondition(String text, boolean use_regex) implements AtmosphericSoundModifier, AtmosphericSoundModifier.Factory {
```

Comment:
By general Java naming conventions, camelCase should be used over snake_case. I would therefore probably name `use_regex` `isRegex`.

Reply:
its a bad habit from lua, yeah, will do